src/clib/pioc_support.c

```diff
@@ -4102,6 +4140,62 @@ static int adios_get_step_info(file_desc_t *file, int varid, size_t adios_step,
     return 0;
 }
 
+static int adios_get_num_decomp_blocks(file_desc_t *file)
+{
+    adios2_variable *variableH = adios2_inquire_variable(file->ioH, "/__pio__/info/block_list");
+    if (variableH == NULL)
+    {
+        return pio_err(NULL, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
+                       "Getting number of decomposition blocks in file (%s, ncid=%d) using ADIOS iotype failed. "
+                       "/__pio__/info/block_list is missing",
+                       pio_get_fname_from_file(file), file->pio_ncid);
+    }
+
+    adios2_varinfo *info_block_list = adios2_inquire_blockinfo(file->engineH, variableH, 0);
+    if (info_block_list == NULL)
+    {
+        return pio_err(NULL, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
+                       "Getting number of decomposition blocks (%s, ncid=%d) using ADIOS iotype failed. "
+                       "The low level (ADIOS) I/O library call failed to get the list of blocks for a variable in a given step (NULL pointer returned)",
+                       pio_get_fname_from_file(file), file->pio_ncid);
+    }
+
+    file->adios_reader_num_decomp_blocks = info_block_list->nblocks;
+
+    /* Free adios2_varinfo structure (this ADIOS2 API returns void) */
+    adios2_free_blockinfo(info_block_list);
+
+    return 0;
```

Comment:
You might want to replace 0 with PIO_NOERR

Reply:
OK, thanks for the suggestion.